documentation/decisions/adr/2024-03-04-logging-monitoring.md

```diff
@@ -0,0 +1,115 @@
+
+- **Decision Status**: Not Recommended
+- **Pros**
+  - Generous free tier
+- **Cons**
+  - Abysmal UX
+
+Links:
+- the first 3 social links for "Cloudwatch UX" are about how bad it is [1](https://news.ycombinator.com/item?id=18584679), [2](https://www.reddit.com/r/aws/comments/nmsapj/this_cloudwatch_ui_sucks/), [3](https://news.ycombinator.com/item?id=18550722)
+- [Pricing](https://aws.amazon.com/cloudwatch/pricing/)
+
+### Sentry
+
+Sentry is a frontend-focused APM (application performance monitoring) platform that has not (yet) expanded to become a logging and metrics platform. It is specifically focused on frontend exception and error handling, as well as performance. As simpler.grants.gov is an API-driven platform, Sentry is a non-ideal choice for our first production operations platform. That said, Sentry fulfills its role very effectively, and with fantastic UX. So Sentry might be a good platform to consider if we decide down the line that we need additional monitoring coverage on the frontend. Sentry is a fairly popular product but is not yet listed by [FedRAMP](https://marketplace.fedramp.gov/products).
+
+- **Decision Status**: Not Recommended (at this time)
+- **Pros**
+  - Great UX
+- **Cons**
+  - Not a fully featured platform
+ 
+### Datadog
+
+Datadog is a fully featured platform with support for logging, metrics dashboards, and APM. Several members of our team have prior experience with Datadog. Datadog excels in the market, is listed on [FedRAMP](https://marketplace.fedramp.gov/products), and would be an excellent choice of platform. It has fantastic UX, for both backend and frontend applications. The only "flaw" is that, at a high level, Datadog's product offering is hard to distinguish from New Relic's. The bulk of this paragraph is copied into the New Relic description, to emphasize that point.
+
+One large advantage of Datadog is that it was built around its metrics and dashboard capabilities, and was built for API-driven applications. As such, Datadog has best-in-class dashboard functionality. This gives it a slight leg-up relative to New Relic. Datadog additionally can make dashboards public, which is a functionality that we may want to leverage to expose our API status (4XX / 5XX rate, etc) to the general public.
+
+Grants.gov used Datadog in the past and then decided to move to Cloudwatch, due to cost.
+
+- **Decision Status**: Top Choice
+- **Pros**
+  - Well-known to our team
+  - Fully featured platform (logs, metrics, APM)
+  - Strong metrics product
+  - Best for API-driven applications
```

Comment:
How does it do with backend applications/scripts? I know with New Relic we were able to configure the API and backend to both work pretty well.

Reply:
It works great - in fact my first usage of Datadog @ my last job was as a part of a ELT script. I just attached the APM / OpenTelemetry reader to the start of the script, and it monitored it just like anything else. I assume New Relic works in exactly the same way.

Reply:
I think for New Relic we had a few utilities for setting up the connection & restructuring our logs, but once that is setup, each ECS task just ended up with `@background_task("WhateverYouWantToCallYourECSTask")` on it and it just worked. Definitely has some setup, but after that it is pretty uneventful

Reply:
For Datadog is looked similar, `@ddtrace.span(name="Whatever")`

There wasn't any meaningful difference between instrumenting a background task versus instrumenting a request. The majority (like 95%) of the work was sitting up the instructure layer, particularly the logging sidecars.